lib/api/apiUtils/object/locationKeysSanityCheck.js

```diff
@@ -0,0 +1,20 @@
+/**
+* Check keys that exist in the current list which will be used in composing
+* object. This method checks against accidentally removing data keys due to
+* instability from the metadata layer. The check returns true if there was no
+* match and false if at least one key from the previous list exists in the
+* current list
+* @param {array|string} prev - list of keys from the object being overwritten
+* @param {array} curr - list of keys to be used in composing current object
+* @returns {array} list of keys that can be deleted
+*/
+function locationKeysSanityCheck(prev, curr) {
+    // backwards compatibility check if object is of model version 2
+    if (typeof prev === 'string') {
+        return curr.every(v => v.key !== prev);
+    }
+    const keysList = prev.map(v => v.key);
```

Comment:
I'm not sure if this case can occur, but we could make an optimization by returning `true` if `keysList.length === 0`.

Reply:
I can add a check at the top.